### PR TITLE
Modify jenkins steps to adjust 4.0 change

### PR DESCRIPTION
**Pull request by xiuwang**
[jenkins log](https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/38703/console)
@openshift/devexp-qe please help review

### CODE REVIEW

**Comment by xiuwang** [comment]
When get server version of 4.0 above cluster, the returned result is "Server Version: version.Info{Major:"1", Minor:"12+".....}", so I add one step "Given /^I have a jenkins server with above 40 cluster$/ do" .
After enable any openshift oauth, we get two choose when login openshift websole, the kubeadmin account and common account. So I add  steps "jenkins_multi_oauth_login" and "Given /^I log into jenkins with above 40 cluster$"

**Comment by akostadinov** [comment]
We should rather update Environment#get_version method work using the 4.x call instead of the old one. Then use existing step /^the master version ([<>=]=?) #{QUOTED}$/

wrt the login step,what is the need of this new step?

**Comment by akostadinov** [comment]
Master branch is only for 4.0 cluster version. Can we update existing step instead?

**Comment by xiuwang** [comment]
@akostadinov since this pr [1], the  version for 4.0 is kubernetes version now, could we use client version instead ?

**Comment by xiuwang** [comment]
For the login step,need add a step to choose the auth type since there are two auth type(kubeadmin and other one auth type) at least.

**Comment by akostadinov** [comment]
I need to check how we can do things. Maybe we can use `clusterversion` instead. WDYT?

**Comment by xiuwang** [comment]
@akostadinov  `get clusterversion` need admin role, IMO, it's not a best choice.
Could we hardcode the version to 4.0 [here](https://github.com/openshift/verification-tests/blob/master/lib/environment.rb#L244) ?  I will try launch a 3.2 version to see if version/openshift could be obtained, later will add more comment.

**Comment by xiuwang** [comment]
Both 3.1 and 3.2 version get 403 forbidden when access version/openshift url.

**Comment by xiuwang** [comment]
We have a cmd `openshift version` that common user could use, this works in all 3.x and 4.0 cluster. 

3.1
# openshift version 
openshift v3.1.1.11-9-g44fe9ba
kubernetes v1.1.0-origin-1107-g4c8e6f4
etcd 2.1.2

4.0
$openshift  version
openshift v4.0.2

**Comment by akostadinov** [comment]
I think that `openshift version` would return version of binary instead of version of server. Am I wrong?

**Comment by xiuwang** [comment]
Alex, you are right, `openshift version` returns the binary version, and this cmd has been deprecated from latest 4.1 package.
IMO, the devel said the way in bug might be  https://bugzilla.redhat.com/show_bug.cgi?id=1658957#c11 .
Meantime ,confirmed with yapei,  `<yapei> xiuwang, no, clusterversion info is shown on Administration -> Cluster Settings page on web console which is invisible for non cluster-admin users`.

**Comment by akostadinov** [comment]
I'm trying to reach directly to the developer to see what he meant by getting api server version.

**Comment by xiuwang** [comment]
[3.11 passed log](https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/39292/console)
[4.0 passed log](https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/39299/console[)

File: features/step_definitions/images/jenkins.rb [hunk 1]
```diff
@@ -102,11 +103,16 @@
 #   non-SSO login.
 # Installation step also installs based on this.
 Given /^I log in to jenkins$/ do
-  if env.version_ge("3.4", user: user) && user.password?
+  if env.version_ge("3.4", user: user) && env.version_le("3.11", user: user) && user.password?
```
Comment (akostadinov) [inline features/step_definitions/images/jenkins.rb]: in this branch we should only support 4.x, conditions for older cluster verisions better be gone. WDYT?

Comment (akostadinov) [reply features/step_definitions/images/jenkins.rb]: not sure what CVP is. In any case, master branch not suitable to run stuff on 3.x

Comment (xiuwang) [reply features/step_definitions/images/jenkins.rb]: CVP is [container verification pipeline tool][https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=CVP&title=Container+Verification+Pipeline+E2E+Documentation#ContainerVerificationPipelineE2EDocumentation-ProductTests) which  anli is taking development.
I will remove the conditions for older cluster verisions later.

**Comment by xiuwang** [comment]
@akostadinov Removed the conditions for older cluster verisions, and test case [passed](https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/39356/consoleFull).

**Comment by akostadinov** [comment]
/lgtm

Thank you!

**Pull request merged**